wrapper/src/main/java/software/amazon/jdbc/PluginService.java

```diff
@@ -62,8 +62,11 @@ EnumSet<NodeChangeOptions> setCurrentConnection(
       @Nullable ConnectionPlugin skipNotificationForThisPlugin)
       throws SQLException;
 
+  // TODO: should we rename to getAllHosts?
   List<HostSpec> getHosts();
```

Comment:
I'd keep getHosts() as-is so we can avoid dozens of changes throughout the code. getHosts() should return a list of nodes excluding those that do not conform with white/disabled lists.

And, at the same time, I'd add another method getAllHosts() that returns all topology nodes including not allowed ones.

Reply:
Okay, so all places that currently call getHosts() on main will now get the subset of hosts? I'm not sure if we want to do that in all places or if some of them should be updated to call getAllHosts()

Reply:
Any particular place where we don't want that subset of hosts to be used?

Reply:
I reviewed the calls to getHosts() with Sergiy and we determined where to grab all the hosts vs just the subset